### ISSUE TITLE
BrainHack 2024

### ISSUE THREAD

**Issue by talmo** [talmolab/ndx-pose#1]
# Overview

**Goal:** Be able to store SLEAP package files in `PoseTraining` objects so we can embed images used for training models together with the ground truth poses.

## Data model equivalences

| `sleap-io` | `ndx-pose` |
|--------|--------|
| `Labels` | `PoseTraining` |
| `LabeledFrame` | `TrainingFrame` |
| `Skeleton` | `Skeleton` |
| `Instance` | `SkeletonInstance` |
| `Video` | `ImageSeries` |
| `Labels.skeletons` | `Skeletons` |
| `Labels.labeled_frames` | `TrainingFrames` |
| `Labels.videos` | `SourceVideos` |

### MVP

- [x] Make `TrainingFrame.skeleton_instance` into `TrainingFrame.skeleton_instances` so it can hold multiple instances per frame
- [ ] Put `SkeletonInstance` creation inside of `TrainingFrame` loop in the README example
- [x] Store image data in [`Image`](https://nwb-schema.readthedocs.io/en/latest/format.html#image) as 3-D array (#2)
- [x] Make `TrainingFrame.source_video` optional so we don't need to keep the source video together with the NWB file (#2)
- [ ] Integrate into [`sleap_io.io.nwb`](https://github.com/talmolab/sleap-io/blob/main/sleap_io/io/nwb.py)
- [ ] Update docs/readme
- [ ] Finish tests

### Nice to have

- [ ] Add a plain string field to `TrainingFrame` to point to source video as provenance but without triggering validation on DANDI
- [ ] Add support for PNG string encoding representation for `Image`
- [ ] Convert and upload [reference SLEAP training datasets](https://sleap.ai/datasets.html) to DANDI
- [ ] Add support for directly opening SLEAP Dandisets in `sleap-io` via [`dandi`](https://dandi.readthedocs.io/en/latest/modref/generated/dandi.download.html#module-dandi.download) API
